rocprofiler-systems build fails with GCC 14: 'any_of' is not a member of 'std'

## Problem Description

Building rocprofiler-systems fails on Ubuntu 25.04 with GCC 14.2 due to missing `#include <algorithm>` in `logger.hpp`.

### Error

```
/develop/therock/rocm-systems/projects/rocprofiler-systems/source/lib/logger/logger.hpp:74:17: error: 'any_of' is not a member of 'std'
```

### Root Cause

`source/lib/logger/logger.hpp` uses `std::any_of` and `std::array` but doesn't include `<algorithm>` or `<array>`. Earlier GCC versions may have included these headers transitively, but GCC 14 requires explicit includes.

### Fix

Add to `logger.hpp`:
```cpp
#include <algorithm>
#include <array>
```

### Environment

- Ubuntu 25.04
- GCC 14.2.0

### Related

Discovered while investigating #3221.

--- a/projects/rocprofiler-systems/source/lib/logger/logger.hpp
+++ b/projects/rocprofiler-systems/source/lib/logger/logger.hpp
@@ -7,6 +7,8 @@
 #include <spdlog/sinks/stdout_color_sinks.h>
 #include <spdlog/spdlog.h>
 
+#include <algorithm>
+#include <array>
 #include <atomic>
 #include <cstdlib>
 #include <mutex>

